libdnf/dnf-sack.cpp

```diff
@@ -2372,8 +2382,14 @@ setModuleExcludes(DnfSack *sack, const char ** hotfixRepos,
     excludeProvidesQuery.addFilter(HY_PKG_PROVIDES, &nameDependencies);
     excludeProvidesQuery.queryDifference(includeQuery);
 
-    // Requred to filtrate out source packages and packages with incompatible architectures
+    // Search for source packages with same names as included source artifacts
+    excludeSrcNamesQuery.addFilter(HY_PKG_NAME, HY_EQ, srcNamesCString.data());
+    const char * srcArchs[] = {"src", "nosrc", nullptr};
+    excludeSrcNamesQuery.addFilter(HY_PKG_ARCH, HY_NEQ, srcArchs);
```

Comment:
The associated CI test doesn't pass for me. In order to make it pass I think here it should be:
```suggestion
    excludeSrcNamesQuery.addFilter(HY_PKG_ARCH, HY_EQ, srcArchs);
```
Since we later do a `queryUnion` we effective exclude all the packages with a name from `srcNamesCString` whose architecture doesn't match "src" or "nosrc", I believe we want the opposite.

Reply:
I am really sorry, I forget to push final and working version. Thanks for note.